sklearn/impute/_base.py

```diff
@@ -649,6 +671,25 @@ def _fit(self, X, y=None):
 
         return missing_features_info[0]
 
+    def _fit_precomputed(self, X, y=None):
+        """Fit the transformer on X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}
+            Mask of input data.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        self._precomputed = False
```

Comment:
This secret attribute that can only be activate by a private function. I am +0.5 with this approach.